relay-general/src/store/clock_drift.rs

```diff
@@ -128,7 +128,7 @@ impl Processor for ClockDriftProcessor {
         if let Some(correction) = self.correction {
             // NB: We're not setting the original value here, as this could considerably increase
             // the event's size. Instead, attach an error message to the top-level event.
-            *timestamp = *timestamp + correction.drift;
+            **timestamp = **timestamp + correction.drift;
```

Comment:
This is fairly indirect. How would you feel about `impl Add<Duration> for Timestamp`? Generally, whenever we can avoid accessing the internal types, we should opt for that.

Probably same goes to expose the inner value via some method.

Reply:
I added an impl for Add but I feel like exposing the inner value is the right call if reexporting all the API surface explicitly isn't really abstracting over anything.

Reply:
That makes sense. In most other cases, it seems need is the chrono `DateTime` instance that's hidden behind a reference to timestamp. A `timestamp.to_date_time()` -- since it is `Copy` -- or `timestamp.to_chrono()` would be more readable than `**x`, even though it's longer.

Over time, I've noticed that Deref isn't all that popular outside of the smart-pointer / ownership wrapper use cases.